### PR TITLE
TW-1702: DownloadErrorPresentationState added handle errors

### PR DESCRIPTION
**Pull request by Te-Z**
Ticket: https://github.com/linagora/twake-on-matrix/issues/1702

Demo:

Web:

[Capture vidéo du 19-04-2024 16:14:52.webm](https://github.com/linagora/twake-on-matrix/assets/31937920/cc3be4b8-7afd-4eaa-b78b-336e5f5699d5)

Mobile:

https://github.com/linagora/twake-on-matrix/assets/31937920/4e4bcb3b-25ef-4bfa-b3ea-2b37795ccade

### CODE REVIEW

File: lib/presentation/model/chat/downloading_state_presentation_model.dart [hunk 1]
```diff
@@ -44,3 +44,12 @@ class DownloadingPresentationState extends DownloadPresentationState {
   @override
   List<Object?> get props => [receive, total];
 }
+
+class DownloadErrorPresentationState extends DownloadPresentationState {
+  final String error;
```
Comment (sherlockvn) [inline lib/presentation/model/chat/downloading_state_presentation_model.dart:49]: i think error should be dynamic

File: lib/widgets/file_widget/download_file_tile_widget.dart [hunk 2]
```diff
@@ -63,11 +65,13 @@ class DownloadFileTileWidget extends StatelessWidget {
                     width: style.iconSize,
                     height: style.iconSize,
                     decoration: BoxDecoration(
-                      color: Theme.of(context).colorScheme.primary,
+                      color: hasError
+                          ? Theme.of(context).colorScheme.error
+                          : Theme.of(context).colorScheme.primary,
                       shape: BoxShape.circle,
```
Comment (sherlockvn) [inline lib/widgets/file_widget/download_file_tile_widget.dart:71]: i think you should put it in `MessageFileTileStyle`

File: lib/widgets/file_widget/download_file_tile_widget.dart [hunk 3]
```diff
@@ -81,13 +85,17 @@ class DownloadFileTileWidget extends StatelessWidget {
                     child: Container(
                       width: style.downloadIconSize,
                       decoration: BoxDecoration(
-                        color: Theme.of(context).colorScheme.primary,
+                        color: hasError
+                            ? Theme.of(context).colorScheme.error
+                            : Theme.of(context).colorScheme.primary,
```
Comment (sherlockvn) [inline lib/widgets/file_widget/download_file_tile_widget.dart:90]: same

File: lib/widgets/file_widget/download_file_tile_widget.dart (hunk 2 above)
Comment (Te-Z) [reply lib/widgets/file_widget/download_file_tile_widget.dart]: done

File: lib/presentation/model/chat/downloading_state_presentation_model.dart (hunk 1 above)
Comment (Te-Z) [reply lib/presentation/model/chat/downloading_state_presentation_model.dart]: done

**Comment by sherlockvn** [comment]
https://github.com/linagora/twake-on-matrix/assets/43041967/d569d9e7-fe9e-4925-a4cf-cbc2fb98197c

When i cancel download, file should be shown normal, not error

**Review by sherlockvn — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**